src/agent/inspectordebugapi.ts

```diff
@@ -312,37 +350,65 @@ export class InspectorDebugApi implements debugapi.DebugApi {
     if (line === 1) {
       column += debugapi.MODULE_WRAP_PREFIX_LENGTH - 1;
     }
-    let condition;
-    if (breakpoint.condition) condition = breakpoint.condition;
-    let res = this.v8Inspector.setBreakpointByUrl(
-        line - 1, matchingScript, undefined, column - 1, condition);
-    if (res.error || !res.response) {
+
+    let result =
+        this.setAndStoreBreakpoint(breakpoint, line, column, matchingScript);
+    if (!result) {
       return utils.setErrorStatusAndCallback(
           cb, breakpoint, StatusMessage.BREAKPOINT_SOURCE_LOCATION,
           utils.messages.V8_BREAKPOINT_ERROR);
     }
+
     this.breakpoints[breakpoint.id as string] = new BreakpointData(
-        res.response.breakpointId, true, breakpoint, ast as estree.Program,
-        compile);
+        result.v8BreakpointId, breakpoint, ast as estree.Program,
+        result.locationStr, compile);
+
     this.numBreakpoints++;
     setImmediate(function() {
       cb(null);
     });  // success.
   }
 
+  private setAndStoreBreakpoint(
+      breakpoint: apiTypes.Breakpoint, line: number, column: number,
+      matchingScript: string):
+      {v8BreakpointId: inspector.Debugger.BreakpointId,
+       locationStr: string}|null {
+    let condition;
+    if (breakpoint.condition) condition = breakpoint.condition;
+    // location Str will be a JSON string of Stackdriver breakpoint location.
+    // It will be used as key at locationmapper to ensure there will be no
+    // duplicate breakpoints at the same location.
+    let locationStr = JSON.stringify(breakpoint.location);
+    let v8BreakpointId;  // v8/inspector breakpoint id
+    if (!this.locationmapper[locationStr]) {
+      // The first time when a breakpoint was set to this location.
+      let res = this.v8Inspector.setBreakpointByUrl(
+          line - 1, matchingScript, undefined, column - 1, condition);
```

Comment:
I'm surprised the compiler doesn't complain that of use before assignment for `condition`. I think you can directly set the last argument to `breakpoint.condition` and remove lines 377-378.

Reply:
Done.